### ISSUE TITLE
False-positive no-value-for-parameter for some unpacked values of list-builtins 

### ISSUE THREAD

**Issue by nelfin** [PyCQA/pylint#4546]
### Steps to reproduce

```python
# pylint: disable=missing-docstring,disallowed-name,invalid-name,trailing-comma-tuple

def foo(arg):
    print(arg)


args = list((1, ))
foo(*args)  # no false-positive


def returns_explicit_tuple(x):
    return tuple([x])


args = list(returns_explicit_tuple(1))
foo(*args)  # no false-positive


def returns_implicit_tuple(x):
    return x,


args = list(returns_implicit_tuple(1))
foo(*args)  # false-positive: no-value-for-parameter

args = returns_implicit_tuple(1)
foo(*args)  # no false-positive
```

### Current behavior

```
************* Module starargs
starargs.py:25:0: E1120: No value for argument 'arg' in function call (no-value-for-parameter)
```

### Expected behavior

`*list(implicit_tuple)` behaviour is equivalent to `*implicit_tuple` behaviour.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.5.dev248+g0e7c2c4
astroid 2.5.7
Python 3.9.0 (default, Nov 30 2020, 15:33:28) 
[GCC 5.4.0 20160609]
```

**Comment by Pierre-Sassoulas** [comment]
I can reproduce with latest pylint/astroid.
```
pylint 2.8.4.dev94+g3ca2b258.d20210607
astroid 2.5.8
Python 3.8.5 (default, May 27 2021, 13:30:53) 
[GCC 9.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
Can still reproduce in 2.15.0-dev0.
